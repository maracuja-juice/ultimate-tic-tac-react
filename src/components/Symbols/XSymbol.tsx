import * as React from 'react';
import classNames from 'classnames';
import styles from './Symbols.module.css';
import { XGameIcon } from '../Icons';

export interface XSymbolProps {
  shouldAnimate?: boolean;
}

export class XSymbol extends React.Component<XSymbolProps> {
  public static defaultProps: Partial<XSymbolProps> = {
    shouldAnimate: true,
  };

  render() {
    const { shouldAnimate } = this.props;
    const iconClass = classNames({
      [styles.x]: true,
      [styles.animateX]: shouldAnimate,
    });

    return <XGameIcon className={iconClass} />;
  }
}
