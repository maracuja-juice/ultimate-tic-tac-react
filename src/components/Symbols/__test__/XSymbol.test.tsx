import * as React from 'react';
import { shallow } from 'enzyme';
import { XSymbol } from '../XSymbol';

describe('XSymbol', () => {
  it('should not explode', () => {
    const component = shallow(<XSymbol />);
    expect(component).not.toBeNull();
  });

  it('should match snapshot', () => {
    const xSymbol1 = shallow(<XSymbol />);
    expect(xSymbol1).toMatchSnapshot();

    const xSymbol2 = shallow(<XSymbol shouldAnimate={false} />);
    expect(xSymbol2).toMatchSnapshot();
  });

  it('should have a child element', () => {
    const component = shallow(<XSymbol />);
    expect(component).toHaveLength(1);
  });

  it('should have x and animateX class by default', () => {
    const component = shallow(<XSymbol />);
    expect(component.hasClass('x')).toBe(true);
    expect(component.hasClass('animateX')).toBe(true);
  });

  it(`shouldn't have animate classes with shouldAnimate=false`, () => {
    const component = shallow(<XSymbol shouldAnimate={false} />);
    expect(component.hasClass('animateX')).toBe(false);
  });
});
