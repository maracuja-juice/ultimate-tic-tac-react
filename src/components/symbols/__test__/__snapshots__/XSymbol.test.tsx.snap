// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XSymbol should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XSymbol
    shouldAnimate={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Cross Symbol",
      "className": "icon-x x animate-x",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Cross Symbol",
        "className": "icon-x x animate-x",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`XSymbol should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XSymbol
    shouldAnimate={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Cross Symbol",
      "className": "icon-x x",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Cross Symbol",
        "className": "icon-x x",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`XSymbol should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XSymbol
    bigSymbol={true}
    shouldAnimate={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Cross Symbol",
      "className": "icon-x x big-symbol animate-x",
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Cross Symbol",
        "className": "icon-x x big-symbol animate-x",
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
